src/main/java/org/broadinstitute/hellbender/utils/codecs/DepthEvidenceBCICodec.java

```diff
@@ -0,0 +1,73 @@
+package org.broadinstitute.hellbender.utils.codecs;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import org.broadinstitute.hellbender.engine.GATKPath;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.sv.DepthEvidence;
+import org.broadinstitute.hellbender.utils.io.BlockCompressedIntervalStream.Reader;
+import org.broadinstitute.hellbender.utils.io.BlockCompressedIntervalStream.Writer;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+public class DepthEvidenceBCICodec extends AbstractBCICodec<DepthEvidence> {
+    private boolean versionChecked = false;
+    private static final String RD_BCI_FILE_EXTENSION = ".rd.bci";
+
+    @Override
+    public DepthEvidence decode( final Reader<DepthEvidence> reader ) throws IOException {
+        if ( !versionChecked ) {
+            if ( !DepthEvidence.BCI_VERSION.equals(reader.getVersion()) ) {
+                throw new UserException("baf.bci file has wrong version: expected " +
+                        DepthEvidence.BCI_VERSION + " but found " + reader.getVersion());
+            }
+            versionChecked = true;
+        }
```

Comment:
This is all completely hidden from the user in the engine classes.  However, the engine classes do the right thing and create a new codec instance for each FeatureReader.  So this will be fine:  Each bci file read will have its version checked just once.